packages/core/src/config.ts

```diff
@@ -37,6 +43,12 @@ export interface Config {
    * @default 'src/App.tsx'
    */
   appShell: `${string}.tsx`;
+
+  /**
+   * Override TypeScript `compilerOptions` for when generating `.d.ts` files
+   * @default '{ incremental: false, noEmitOnError: false }'
+   */
+  dtsOptions: Record<string, unknown>;
```

Comment:
This is typed as it is and not `ts.CompilerOptions` because `rollup-plugin-dts` fails when bundling the `multi-entry-library` fixture (which references Crackle types).

```
From Rollup plugin dts:
Error: namespace must have a "ModuleBlock" body.

  5544 |     type InvalidatedProject<T extends BuilderProgram> = UpdateOutputFileStampsProject | BuildInvalidedProject<T> | UpdateBundleProject<T>;
  5545 | }
> 5546 | declare namespace ts.server {
       | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 5547 |     type ActionSet = "action::set";
       | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 5548 |     type ActionInvalidate = "action::invalidate";
       | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 5549 |     type ActionPackageInstalled = "action::packageInstalled";
       | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 5550 |     type EventTypesRegistry = "event::typesRegistry";
       | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 5551 |     type EventBeginInstallTypes = "event::beginInstallTypes";
       | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 5552 |     type EventEndInstallTypes = "event::endInstallTypes";
       | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 5553 |     type EventInitializationFailed = "event::initializationFailed";
       | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 5554 | }
       | ^^
  5555 | declare namespace ts.server {
  5556 |     interface TypingInstallerResponse {
  5557 |         readonly kind: ActionSet | ActionInvalidate | EventTypesRegistry | ActionPackageInstalled | EventBeginInstallTypes | EventEndInstallTypes | EventInitializationFailed;
    at DeclarationScope.convertNamespace (/crackle/node_modules/.pnpm/rollup-plugin-dts@4.2.2_rollup@2.78.1/node_modules/rollup-plugin-dts/dist/rollup-plugin-dts.cjs:1078:19)
    at Transformer.convertNamespaceDeclaration (/crackle/node_modules/.pnpm/rollup-plugin-dts@4.2.2_rollup@2.78.1/node_modules/rollup-plugin-dts/dist/rollup-plugin-dts.cjs:1270:15)
    ...
```